tests/components/huum/test_config_flow.py

```diff
@@ -0,0 +1,112 @@
+"""Test the huum config flow."""
+from unittest.mock import patch
+
+import aiohttp
+
+from homeassistant import config_entries, setup
+from homeassistant.components.huum.config_flow import CannotConnect
+from homeassistant.components.huum.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    TEST_USERNAME = "test-username"
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    with patch(
+        "homeassistant.components.huum.config_flow.Huum.status",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.huum.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": TEST_USERNAME,
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == TEST_USERNAME
+    assert result2["data"] == {
+        "username": TEST_USERNAME,
+        "password": "test-password",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.huum.config_flow.Huum.status",
+        side_effect=aiohttp.ClientError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
```

Comment:
Please finish config flows in either Create entry or abort so we also test that the config flow can recover from an error

Reply:
Sorry, but I don't fully grasp what you are after here. Could you please point me in the right direction?

Reply:
Well, after testing that you are redirected back to the form, we should test that the form is still able to complete. So what you do is you patch the lib again to succeed and submit again

Reply:
Aaah, alright, sorry, then I understand what you mean 👍  I rather ask stupid questions here than change the code over and over, better to get it at least close to the right thing on the first try 😄 Thank you once again 🙇

Reply:
No problem :)

Reply:
Added to the tests 👍